### ISSUE TITLE
Heading in degrees is not shown, but compass is when GPS unavailable

### ISSUE THREAD

**Issue by ABLomas** [iNavFlight/inav#8614]
Setup: fixed wing without GPS, but with MAG, analog OSD, 5.1 and 6.0FP1.

There are two separate elements on OSD, which show heading (one "graphical", another numerical (+ offset IIRC):
![photo_2022-12-06_23-15-18](https://user-images.githubusercontent.com/8989151/206121867-81785bae-cc2f-402f-962f-e250fd5d6b4c.jpg)
But only one is shown on actual OSD - graphical version, not numeric. Numeric version is not shown at all, not even dashes - empty space in it's place.
Also confirmed by Igor at telegram chat.

Quick look into code, looks like both elements do not depend on GPS:
`#endif // GPS
    OSD_ELEMENT_ENTRY("HEADING", OSD_HEADING),
    OSD_ELEMENT_ENTRY("HEADING GR.", OSD_HEADING_GRAPH),`

Yet only heading graph is shown, empty space for other (but it's selectable on configurator, while other GPS related elements are hidden).
AFAIK heading should be known to FC using MAG, so what's the problem to display it on screen?

**Comment by breadoven** [comment]
The code above is just for the CMS display so doesn't affect what's actually drawn. However, the OSD code itself that does determine what's drawn does make the numeric heading dependent on GPS, but not the heading graph. This doesn't appear to be correct since heading is valid if you have a compass. Needs fixing.

**Comment by breadoven** [comment]
Actually the `defines` just affect things at compile time but you're using release versions rather than custom built versions with GPS removed so GPS will be defined. You've just switched GPS off in Configurator which removes it as a feature, however this shouldn't cause the OSD numeric heading display to disappear which indeed it does (I checked). It should at least show "---" if the heading isn't valid but it's missing entirely as are other elements such as Home Distance etc.

Checking again I think this is probably caused by something I noticed a while ago and was intending to check and fix but I never got around to it:
https://github.com/iNavFlight/inav/blob/91b4e6eda628fd0eafcebd5821da4dca7d144d9f/src/main/io/osd.c#L3221

Appears this wasn't updated correctly/at all as OSD elements were changed ... looks like it finally needs updating.
